Issue: #7648: add doc for magic number

paragpachpute opened the pull request:
Issue: #7648: add doc for magic number

Screenshot of the website
![Screen Shot 2020-03-17 at 9 59 20 PM](https://user-images.githubusercontent.com/8281380/76918131-f05bf780-689b-11ea-94ec-7acefe29d4ef.png)

![Screen Shot 2020-03-19 at 2 38 31 PM](https://user-images.githubusercontent.com/8281380/77102531-5e650380-69ef-11ea-8204-8412087e5694.png)

Screenshot of the default behavior:
![Screen Shot 2020-03-19 at 2 11 20 PM](https://user-images.githubusercontent.com/8281380/77101907-6cfeeb00-69ee-11ea-955b-557d106ef80c.png)

Screenshot after adding the config:
![Screen Shot 2020-03-18 at 12 28 57 PM](https://user-images.githubusercontent.com/8281380/76983631-1f15b480-6914-11ea-934a-2524071eac2f.png)

paragpachpute commented:
Screenshot of the website
![Screen Shot 2020-03-17 at 9 59 20 PM](https://user-images.githubusercontent.com/8281380/76918131-f05bf780-689b-11ea-94ec-7acefe29d4ef.png)

romani commented:
please update PR description , it is editable. Please move all required content to PR description.
Please read section "PLEASE READ" in issue and follow it. If still not clear what to do, review merged PRs for similar update on documentation.

paragpachpute commented:
> please update PR description , it is editable. Please move all required content to PR description.
> Please read section "PLEASE READ" in issue and follow it. If still not clear what to do, review merged PRs for similar update on documentation.

Updated PR description and added necessary screenshots

romani commented on src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MagicNumberCheck.java:
```diff
@@ -212,6 +212,25 @@
+ * <pre>
+ * class TestHashCode {
+ *     public int hashCode() {
+ *         return 10;       // OK
```
good, but please add `hashCode` to default example to show that by default it will be violation.
But with  `ignoreHashCodeMethod` - there are no violation.

paragpachpute replied:
addressed the comment
